ext/session/session.stub.php

```diff
@@ -80,13 +80,13 @@ public function write(string $id, string $data);
     /** @return bool */
     public function destroy(string $id);
 
-    /** @return int|bool */
+    /** @return int|false */
     public function gc(int $max_lifetime);
 }
 
 interface SessionIdInterface
 {
-    /** @return string */
+    /** @return string|false */
```

Comment:
This doesn't look right to me, in that the create_sid caller will throw an `No session id returned by function` Error for any non-string return value.

Reply:
I don't get what the correct solution should be. If we changed the return type of `SessionHandler::create_sid()` to `string`, then an exception should be thrown (instead of returning `false`) when the session is not active in order to have a compatible return type with the interface. Is it acceptable to do?

Reply:
> Is it acceptable to do?

I guess so, if we end up with an exception anyway.

Reply:
> > Is it acceptable to do?
> 
> I guess so, if we end up with an exception anyway.

Yes, that.

Possibly we can change this error condition in general -- if I understand right, these method are only supposed to be called by the session implementation, and the pre-conditions should already be established in that case. But I'm not certain on this point.

Reply:
I had a look around.. and it seems that indeed neither of the `SessionHandler` methods can be called by the session implementation without an active session state. So I'm wondering if we could simply drop the `if (PS(session_status) != php_session_active)` condition, rather than promoting it to an exception? I'm not sure which is less intrusive of a change. :D